app-backend/database/src/main/scala/com/azavea/rf/database/tables/Shapes.scala

```diff
@@ -0,0 +1,178 @@
+}
+
+object Shapes extends TableQuery(tag => new Shapes(tag)) with LazyLogging {
+
+  val tq = TableQuery[Shapes]
+  type TableQuery = Query[Shapes, ShapeModel, Seq]
+
+
+  implicit class withShapesTableQuery[M, U, C[_]](shapes: Shapes.TableQuery) extends
+    ShapeTableQuery[M, U, C](shapes)
+
+  /** List shapes given a page request
+    *
+    * @param offset Int offset of request for pagination
+    * @param limit Int limit of objects per page
+    * @param queryParams ShapeQueryparameters query parameters for request
+    */
+  def listShapes(offset: Int, limit: Int, queryParams: ShapeQueryParameters, user: User) = {
+
+    val dropRecords = limit * offset
+    val filteredShapes = Shapes
+      .filterByUser(queryParams.userParams)
+    val pagedShapes = filteredShapes
+         .drop(dropRecords)
+         .take(limit)
+
+    ListQueryResult[ShapeModel](
+      (pagedShapes.result):DBIO[Seq[ShapeModel]],
+      filteredShapes.length.result
+    )
+  }
+
+  /** Insert a Shape given a create case class with a user
+    *
+    * @param shapeToCreate Shape.Create object to use to create full Shape
+    * @param user               User to create a new Shape with
+    */
+  def insertShape(shapeToCreate: ShapeModel.Create, user: User) = {
+    val shape = shapeToCreate.toShape(user)
+    (Shapes returning Shapes).forceInsert(shape)
+  }
+
+  /** Insert many Shapes given a secreate case class with a user
+    *
+    * @param shapesToCreate Seq[Shape.Create] to use to create full Shape
+    * @param user                User to create a new Shape with
+    */
+  def insertShapes(shapesToCreate: Seq[ShapeModel.Create], user: User)
+                       (implicit database: DB) = {
+    val shapesToInsert = shapesToCreate map { _.toShape(user) }
+    database.db.run {
+      Shapes.forceInsertAll(
+        shapesToInsert
+      )
+    } map { res =>
+      shapesToInsert.map(_.toGeoJSONFeature)
+    }
+  }
+
+
+  /** Given a Shape ID, attempt to retrieve it from the database
+    *
+    * @param shapeId UUID ID of shape to get from database
+    * @param user     Results will be limited to user's organization
+    */
+  def getShape(shapeId: UUID, user: User) =
+    Shapes
+      .filterToSharedOrganizationIfNotInRoot(user)
+      .filter(_.id === shapeId)
+      .result
+      .headOption
+
+  /** Given a Shape ID, attempt to remove it from the database
+    *
+    * @param shapeId UUID ID of shape to remove
+    * @param user     Results will be limited to user's organization
+    */
+  def deleteShape(shapeId: UUID, user: User) =
+    Shapes
+      .filterToSharedOrganizationIfNotInRoot(user)
+      .filter(_.id === shapeId)
+      .delete
+
+  /** Given a Project ID, attempt to delete all associated shapes from the database
+    *
+    * @param user     Results will be limited to user's organization
+    */
+  def deleteProjectShapes(user: User) =
```

Comment:
Going to remove this as it's unused and not needed. Also it wouldn't do what it claims to do.